### PR TITLE
fix: fix batch mutation limit

### PR DESCRIPTION
**Pull request by mutianf**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/BulkMutation.java [hunk 1]
```diff
@@ -69,6 +69,9 @@ private void writeObject(ObjectOutputStream output) throws IOException {
   public BulkMutation add(@Nonnull String rowKey, @Nonnull Mutation mutation) {
     Preconditions.checkNotNull(rowKey);
     Preconditions.checkNotNull(mutation);
+    Preconditions.checkArgument(
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/BulkMutation.java:72]: do you want to also check the total mutation count across rows as well?

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/mutaterows/MutateRowsBatchResource.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  static MutateRowsBatchResource create(long elementCount, long byteCount, long mutationCount) {
+    return new AutoValue_MutateRowsBatchResource(elementCount, byteCount, mutationCount);
+  }
+
+  @Override
+  public BatchResource add(BatchResource batchResource) {
+    Preconditions.checkArgument(
+        batchResource instanceof MutateRowsBatchResource,
+        "Expected MutateRowsBatchResource, got " + batchResource.getClass());
+    MutateRowsBatchResource mutateRowsResource = (MutateRowsBatchResource) batchResource;
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/mutaterows/MutateRowsBatchResource.java:38]: nit. please add a newline to make it easier to read

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/BulkMutation.java (hunk 1 above)
Comment (mutianf) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/BulkMutation.java:72]: Hmm I'll need to iterate through all the entries I added to BulkMutation and calculate the sum everytime I add some elements. Is it worth? I think as long as someone is using the batcher this should be handled by the batcher?

Comment (igorbernstein2) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/BulkMutation.java:75]: why not keep a running sum?

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/BulkMutation.java [hunk 3]
```diff
@@ -81,6 +86,11 @@ public BulkMutation add(@Nonnull ByteString rowKey, @Nonnull Mutation mutation)
     Preconditions.checkNotNull(rowKey);
     Preconditions.checkNotNull(mutation);
 
+    this.mutationCountSum += mutation.getMutations().size();
+    Preconditions.checkArgument(
+        mutationCountSum <= MAX_MUTATION,
+        String.format("Too many mutations, got %s, limit is %s", mutationCountSum, MAX_MUTATION));
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/BulkMutation.java:92]: you need to check the precondition before modifying state

**Review by igorbernstein2 — APPROVED** [review]

**Comment by mutianf** [comment]
I removed all the checks in client because of https://google.aip.dev/client-libraries/4223

**Pull request merged**